lib/java_buildpack/framework/sealights_agent.rb

```diff
@@ -0,0 +1,168 @@
+# frozen_string_literal: true
+
+# Cloud Foundry Java Buildpack
+# Copyright 2013-2020 the original author or authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'java_buildpack/component/versioned_dependency_component'
+require 'java_buildpack/framework'
+require 'uri'
+require 'net/http'
+require 'pathname'
+require 'base64'
+require 'json'
+
+module JavaBuildpack
+  module Framework
+
+    # Encapsulates the functionality for enabling zero-touch Sealights support.
+    class SealightsAgent < JavaBuildpack::Component::BaseComponent
+
+      def initialize(context)
+        super(context)
+        @logger = JavaBuildpack::Logging::LoggerFactory.instance.get_logger SealightsAgent
+      end
+
+      # (see JavaBuildpack::Component::BaseComponent#detect)
+      def detect
+        supports? ? "#{self.class.to_s.dash_case}=latest" : nil
+      end
+
+      # (see JavaBuildpack::Component::BaseComponent#compile)
+      def compile
+        announce("'compile' method is starting")
+        download_url = get_download_url
+        download_uri(download_url)
+        extract_zip('sealights-java.zip', get_agent_path)
+        announce("'compile' method has ended")
+      end
+
+      def get_download_url
```

Comment:
Can you explain how you typically download the agent for Sealights? It seems like the bound service would contain a URL and the agent can be obtained through that URL. Is it possible to download the agent in advance or is the downloaded agent customized in some way/unique to the individual downloading it? 

Also, how would this work in an environment where there is no access to the Internet? Is it possible for Sealights to be hosted on the LAN, so that the server you'd be taking to here would also be on the LAN? Or is the Internet a requirement for the product? 

Thanks for clarifying

Reply:
Outside of PCF, our customers either downloading the agent using `wget` with token or just make sure that it will be present next to their application. 

Per your question, whether the agent can be downloaded in advance if by advance you mean to be part of the buildpack then yes, also the preferred way is to download it as it allows us to upgrade/downgrade per customer.

Regarding access to the internet, we need to send data to our cloud, thus that's mandatory. That being, we have one prospect that mentioned that he can send data to the internet but can't download anything during the buildpack due to their policy. For this latter case, I'm not sure what's the preferred way of solving that? Should I include a default agent as part of the buildpack?